di2f/index.md

```diff
@@ -0,0 +1,226 @@
+
+# Thread DIDs
+
+<!-- @sanderpick What are we trying to "solve" or say in this section specifically? Why, in one or two sentences, do we need to build this? -->
+
+In practice, a thread _network_ may have multiple actors and structures (subjects) that can be described by DID documents. Subjects include any of the following entities:
+
+Peers
+: _Peers_ (e.g., `did:p2p:foo`) can offer network services, such as thread "hosting", pinning services (i.e., IPFS, IPNS), Filecoin anchoring, database abstractions, API services, and more. A thread peer's DID is derived from its embedded networking host's key (which in practice is a `libp2p` ([`https://libp2p.io/`](https://libp2p.io/)) peer]).
+
+Users
+: A _User_ (e.g., `did:key:foo`, `did:3:foo`, `did:ethr:foo`, etc.) is any external identity that represents a network user, and that may interact with the network _via_ a Peer. These may be identified by any verifiable DID.
+
+Thread:
+: _Thread DID_ (e.g., `did:thread:id`) documents contain verification methods for all valid log signers. Other thread info such as the _log head_, _log metadata_, and _thread encryption keys_ are _not_ stored in the DID document, as this information is only needed by peers that are sharing a thread, and can be more efficiently exchanged using the thread protocol [@pick2020protocol] directly (vs. a global document registry). _Log addresses_ are referenced as _service endpoints_, as defined in [@reed2020decentralized] sec 5.4.
+
+<!-- @sanderpick Rather than introducing a new term here (log signers), could we just say for all valid peers and/or users? -->
+
+By identifying a thread as a global resource, any peer can determine the following from its DID:
+
+1.  Who can write to the thread?
+2.  Who can read from the thread?
+3.  Where can the thread be bootstrapped from?
+4.  Who controls (1), (2), and (3)?
+
+It is important to note that a thread DID is an _identifier_, not an _identity_ in the usual sense. To illustrate, if a thread is a resource that a peer on the network is attempting to identify, i.e., the thread itself _is the subject_, and the thread document is a _representation of the subject_, then the DID _subject is not the controller_ [@reed2020decentralized]. The controller is one or more identity-based DIDs, e.g., `did:key:foo`, `did:ethr:foo`, `did:3:foo`, etc.
+
+A (proposed) thread DID document is structured as in [@lst:thread-did]. In this example, the controller (`did:key:foo`) is defined by a key-based DID [@longely2021didKey]. The controller is able to modify the thread DID document. Additionally, both `did:key:foo"` and `did:key:bar"` can authenticate as `did:thread:id`, meaning ... . Finally, the thread DID document in [@lst:thread-did] also specifies two `serviceEndpoint`s that outlines which peers (`/p2p/peer-id-1` and `/p2p/peer-id-2`) are able to bootstrap the thread.
+
+<!-- @sanderpick What does "authenticate as `"did:thread:id"`" mean in this context? Can we be more explicit here" -->
```

Comment:
@sanderpick What does "authenticate as `"did:thread:id"`" mean in this context? Can we be more explicit here?

Reply:
Oh, using "authentication" as a kind of ACL... those are they valid log singers mentioned above

Reply:
ok, so something like: "meaning they are part of the access control list (ACL), or have "controlled access" to the thread, allowing them to sign and append records to a thread log directly, or via a thread peer.